src/mixins/segmentation/circleSegmentationMixin.js

```diff
@@ -0,0 +1,167 @@
+ *
+ * @param {Object} evt Cornerstone.event#cornerstoneimagerendered > cornerstoneimagerendered event
+ * @memberof Tools.ScissorsTool
+ * @returns {void}
+ */
+function renderToolData(evt) {
+  const eventData = evt.detail;
+  const { element } = eventData;
+  const color = toolColors.getColorIfActive({ active: true });
+  const context = getNewContext(eventData.canvasContext.canvas);
+  const { distance } = external.cornerstoneMath.point;
+
+  draw(context, context => {
+    if (!this.handles) {
+      return null;
+    }
+    const startCanvas = external.cornerstone.pixelToCanvas(
+      element,
+      this.handles.start
+    );
+
+    const endCanvas = external.cornerstone.pixelToCanvas(
+      element,
+      this.handles.end
+    );
+
+    // Calculating the radius where startCanvas is the center of the circle to be drawn
+    const radius = distance(startCanvas, endCanvas);
+
+    // Draw Circle
+    drawCircle(
+      context,
+      element,
+      this.handles.start,
+      radius,
+      {
+        color,
+      },
+      'pixel'
+    );
+  });
+}
+
+/**
+ * Sets the start handle point and claims the eventDispatcher event
+ *
+ * @private
+ * @param {*} evt // mousedown, touchstart, click
+ * @returns {Boolean} True
+ */
+function _startOutliningRegion(evt) {
+  const consumeEvent = true;
+  const element = evt.detail.element;
+  const image = evt.detail.currentPoints.image;
+
+  if (_isEmptyObject(this.handles.start)) {
+    this.handles.start = image;
+  } else {
+    this.handles.end = image;
+    this._applyStrategy(evt);
+  }
+
+  external.cornerstone.updateImage(element);
+
+  return consumeEvent;
+}
+
+/**
+ * This function will update the handles and updateImage to force re-draw
+ *
+ * @private
+ * @method _setHandlesAndUpdate
+ * @param {(CornerstoneTools.event#TOUCH_DRAG|CornerstoneTools.event#MOUSE_DRAG|CornerstoneTools.event#MOUSE_MOVE)} evt  Interaction event emitted by an enabledElement
+ * @returns {void}
+ */
+function _setHandlesAndUpdate(evt) {
+  const {
+    element,
+    currentPoints: { image },
+  } = evt.detail;
+
+  this.handles.end = image;
+  external.cornerstone.updateImage(element);
+}
+
+/**
+ * Event handler for MOUSE_UP/TOUCH_END during handle drag event loop.
+ *
+ * @private
+ * @method _applyStrategy
+ * @param {(CornerstoneTools.event#MOUSE_UP|CornerstoneTools.event#TOUCH_END)} evt Interaction event emitted by an enabledElement
+ * @returns {void}
+ */
+function _applyStrategy(evt) {
+  evt.detail.handles = this.handles;
+  const { image, element } = evt.detail;
+
+  const {
+    labelmap3D,
+    currentImageIdIndex,
+    activeLabelmapIndex,
+  } = getters.getAndCacheLabelmap2D(element);
+
+  const points = [
+    {
+      x: this.handles.start.x,
+      y: this.handles.start.y,
+    },
+    {
+      x: this.handles.end.x,
+      y: this.handles.end.y,
+    },
+  ];
+
+  const toolData = getters.labelmapBuffers(element, activeLabelmapIndex);
```

Comment:
This API returns a whole 3D buffer, and is more intended for interfacing with other libraries.

Reply:
Fixed.